src/fromager/sources.py

```diff
@@ -143,6 +161,10 @@ def download_url(destination_dir: pathlib.Path, url: str) -> pathlib.Path:
     # Open the URL first in case that fails, so we don't end up with an empty file.
     logger.debug(f"reading from {url}")
     with requests.get(url, stream=True) as r:
+        if r.status_code != 200:
```

Comment:
I think you can just call `raise_for_status()` here. https://requests.readthedocs.io/en/latest/user/quickstart/#response-status-codes